android/engine/src/main/java/org/smartregister/fhircore/engine/ui/questionnaire/QuestionnaireActivity.kt

```diff
@@ -65,26 +67,31 @@ import timber.log.Timber
  * Implement a subclass of this [QuestionnaireActivity] to provide functionality on how to
  * [handleQuestionnaireResponse]
  */
+enum class QuestionnaireType {
+  READ_ONLY,
+  EDIT,
+  DEFAULT;
+
+  fun isEditMode() = this == EDIT
+  fun isReadOnly() = this == READ_ONLY
+}
+
```

Comment:
We can move this to the same package level where we have the `QuestionnaireActivity`